eng/native/configurecompiler.cmake

```diff
@@ -577,4 +577,18 @@ else (CLR_CMAKE_HOST_WIN32)
     if (AWK STREQUAL "AWK-NOTFOUND")
         message(FATAL_ERROR "AWK not found")
     endif()
+
+    # detect linker
+    set(ldVersion ${CMAKE_C_COMPILER};-Wl,--version)
+    execute_process(COMMAND ${ldVersion}
+        ERROR_QUIET
+        OUTPUT_VARIABLE ldVersionOutput)
+
+    if("${ldVersionOutput}" MATCHES "GNU ld")
+        set(LD_GNU 1)
+    elseif("${ldVersionOutput}" MATCHES "GNU gold")
+        set(LD_GOLD 1)
```

Comment:
I think we should identify both ld and ld gold as LD_GNU. We compile with ld gold explicitly when CLR_CMAKE_PGO_INSTRUMENT is set.

Reply:
Merged gold and GNU branches.